### PR TITLE
Add offline support

### PR DESCRIPTION
**Pull request by lpusok**
### Checklist

- [x] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-io/bitrise/blob/master/.github/CONTRIBUTING.md)
- [ ] `README.md` is updated with the changes (if needed)

### Version

Requires a *MINOR* [version update](https://semver.org/)

### Context

Added *steps* command with subcommands:
```
  list-cached     List all the cached steps
  preload         Makes sure that Bitrise CLI can be used in offline mode by preloading Bitrise maintaned Steps.
```

The preload can be configured like this:
```
NAME:
   bitrise steps preload - Makes sure that Bitrise CLI can be used in offline mode by preloading Bitrise maintaned Steps.

USAGE:
   Use the BITRISE_OFFLINE_MODE env var to test after preloading steps.

OPTIONS:
   --steplib-url value    URL of the steplib to list or preload steps from (default: "https://github.com/bitrise-io/bitrise-steplib.git")
   --maintainer value     Maintainer of the steps to list or preload (default: "bitrise")
   --majors value         Include X latest major versions (default: 2)
   --minors value         Include X latest minor versions for each major version (default: 1)
   --minors-since value   Include latest patch version of minors that were released in the last X months (default: 2)
   --patches-since value  Include all patch version that were released in the last X months (default: 1)
```

### CODE REVIEW

File: bitrise/setup.go [hunk 1]
```diff
@@ -77,12 +80,17 @@ func RunSetup(appVersion string, setupMode SetupMode, doCleanSetup bool) error {
 		return fmt.Errorf("Failed to do common/platform independent setup, error: %s", err)
 	}
 
+	deps := []string{}
```
Comment (ofalvai) [inline bitrise/setup.go:83]: This is empty, are you sure it's needed?

File: cli/preload_steps.go [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+	bitriseStepLibURL = "https://github.com/bitrise-io/bitrise-steplib.git"
+	bitriseMaintainer = "bitrise"
+)
+
+var stepsCommand = cli.Command{
+	Name:  "steps",
+	Usage: "Manage Steps cache.",
+	Subcommands: []cli.Command{
+		{
+			Name:  "list",
```
Comment (ofalvai) [inline cli/preload_steps.go:24]:
```suggestion
			Name:  "list-cached",
```
It's a bit confusing without knowing about offline mode

File: bitrise/setup.go (hunk 1 above)
Comment (lpusok) [reply bitrise/setup.go:83]: Not needed right now, added this to install zstd. Should I restore all setup.go cahnges?

File: cli/preload_steps.go (hunk 2 above)
Comment (lpusok) [reply cli/preload_steps.go]: Renamed.

File: cli/setup.go [hunk 3]
```diff
@@ -19,10 +19,6 @@ var setupCommand = cli.Command{
 		return nil
 	},
 	Flags: []cli.Flag{
-		cli.BoolFlag{
```
Comment (lpusok) [inline cli/setup.go:22]: Not used anywhere.

**Review by ofalvai — APPROVED** [review]

**Pull request merged**